starknet-rpc-test/src/lib.rs

```diff
@@ -20,10 +30,22 @@ use url::Url;
 pub struct MadaraClient {
     process: Child,
     client: Client,
-    rpc_request_count: Cell<usize>,
+    rpc_request_count: AtomicUsize,
     starknet_client: JsonRpcClient<HttpTransport>,
 }
 
+#[derive(Debug, Default)]
+pub struct BlockCreation {
+    parent_hash: Option<String>,
+    finalize: bool,
+}
+
+impl BlockCreation {
+    pub fn new(parent_hash: Option<String>, finalize: bool) -> Self {
+        BlockCreation { parent_hash, finalize }
+    }
+}
+
```

Comment:
```rust
madara
        .create_block(
            vec![transfer_tokens(
                &account,
                argent_account_address,
                FieldElement::from_hex_be(MINT_AMOUNT).expect("Invalid Mint Amount"),
                None,
            )],
            BlockCreation::new(None, true),
        )
        .await?;
```

I find it to be poorly expressive. When calling I don't know what is `None` and what this `bool` stand for. 
I would rather have no `new` function so that the caller can use:

```rust
BlockCreation {
    parent_hash: None,
    finalize: true,
}
```

This way we know what is going on